aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -135,30 +135,43 @@ Arbitrary items are supported as content of the `<NotFound>` tags, such as other
 
 ## Route to components from multiple assemblies
 
-Use the <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies> parameter to specify additional assemblies for the <xref:Microsoft.AspNetCore.Components.Routing.Router> component to consider when searching for routable components. Additional assemblies are scanned in addition to the assembly specified to `AppAssembly`. In the following example, `Component1` is a routable component defined in a referenced [component class library](xref:blazor/components/class-libraries). The following <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies> example results in routing support for `Component1`.
+Use the <xref:Microsoft.AspNetCore.Components.Routing.Router> component's <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies> parameter to discover routable components in additional assemblies. Additional assemblies are scanned in addition to the assembly specified to <xref:Microsoft.AspNetCore.Components.Routing.Router.AppAssembly%2A>.
 
-:::moniker range=">= aspnetcore-6.0"
+In the following example, `Component1` is a routable component defined in a referenced [component class library](xref:blazor/components/class-libraries) named `ComponentLibrary`:
 
 ```razor
 <Router
-    AppAssembly="@typeof(App).Assembly"
-    AdditionalAssemblies="new[] { typeof(Component1).Assembly }">
-    @* ... Router component elements ... *@
+    AppAssembly="..."
+    AdditionalAssemblies="new[] { typeof(ComponentLibrary.Component1).Assembly }">
+    ...
 </Router>
 ```
 
-:::moniker-end
+:::moniker range=">= aspnetcore-8.0"
 
-:::moniker range="< aspnetcore-6.0"
+In the following Blazor Web App example, the `_Imports.razor` file of the `BlazorSample.Client` project indicates the assembly to search for routable components:
 
 ```razor
 <Router
-    AppAssembly="@typeof(Program).Assembly"
-    AdditionalAssemblies="new[] { typeof(Component1).Assembly }">
-    @* ... Router component elements ... *@
+    AppAssembly="..."
+    AdditionalAssemblies="new[] { typeof(BlazorSample.Client._Imports).Assembly }">
+    ...
 </Router>
 ```
 
+A Blazor Web App is required to indicate the `.Client` project's assembly to <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies> if the Blazor router is interactive, either on the server or on the client.
+
+If the the Blazor router isn't an interactive router, routable components of the `.Client` project are discovered without specifying the `.Client` project's assembly to <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies>.
+
+To discover components from additional assemblies for static server-side rendering, identify the additional assemblies with the <xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointConventionBuilderExtensions.AddAdditionalAssemblies%2A> method chained to <xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointRouteBuilderExtensions.MapRazorComponents%2A>.
```

Comment:
This first paragraph makes it sound to me like, if the router is not interactive, you don't have to do anything to have the `.Client` project's routable components get discovered. Then the second paragraph indicates that you _do_ have to do something, and that's registering the additional assemblies using `AddAdditionalAssemblies`. I wonder if there's a way to unify those somehow. Or if you think it's clear enough already, no need to try to reword it!

I've also just written up an example below of how I would explain the concept of "additional assemblies" as it pertains to various app configurations. Feel free to ignore it completely if you like what you have, but I'm putting it here in case you want to borrow any ideas from it:

---

If the router is interactive, any additional assemblies containing routable components must be listed in <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies>.

To have routable components from additional assemblies get discovered for static server rendering, those assemblies must be listed in <xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointConventionBuilderExtensions.AddAdditionalAssemblies%2A>.

For example, in a Blazor Web app, an interactive render mode can be specified on the `Routes` component, which in turn renders the `Router` component. If the `Routes` component is defined in the server project, the <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies> parameter should include the `.Client` assembly. This allows the router to work correctly even when rendered interactively.

However, in a Blazor Web app with global WebAssembly interactivity enabled, the `Routes` component might be defined in the `.Client` project. In this case, there aren't assemblies external to the one defining the `Routes` component that contain routable components, so it's not necessary to specify a value for <xref:Microsoft.AspNetCore.Components.Routing.Router.AdditionalAssemblies>.

In both cases, though, it's necessary to include routable components in the `.Client` assembly for static server rendering using <xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointConventionBuilderExtensions.AddAdditionalAssemblies%2A>. This is because the initial, static server-side render needs to be able to identify components in the `.Client` assembly, even if the router later becomes interactive.

Reply:
That's ***much*** better. I'll place your text for the coverage.

WRT "static" versus "interactive" routers: Can you describe them for me?

Reply:
Nice ... I'll roll this to the new section at the top of the article first thing Wednesday morning.